### PR TITLE
[cdc-connector][Postgres][hotfix] Fix postgresSQL database connection leak

### PR DESCRIPTION
**Pull request by loserwang1024**
Fix https://github.com/ververica/flink-cdc-connectors/issues/2980. Now that PostgresSourceFetchTaskContext can only be contained by single thread now, it is no need to put into AtomicReference.

### CODE REVIEW

**Comment by loserwang1024** [comment]
@leonardBang , @ruanhang1993 , CC

File: flink-cdc-connect/flink-cdc-source-connectors/flink-connector-postgres-cdc/src/main/java/com/ververica/cdc/connectors/postgres/source/fetch/PostgresSourceFetchTaskContext.java [hunk 1]
```diff
@@ -302,8 +301,11 @@ public Offset getStreamOffset(SourceRecord sourceRecord) {
     }
 
     @Override
-    public void close() {
+    public void close() throws Exception {
         jdbcConnection.close();
+        if (replicationConnection != null) {
```
Comment (leonardBang) [inline flink-cdc-connect/flink-cdc-source-connectors/flink-connector-postgres-cdc/src/main/java/com/ververica/cdc/connectors/postgres/source/fetch/PostgresSourceFetchTaskContext.java:306]: check null for `jdbcConnection` as well ?

Comment (loserwang1024) [reply flink-cdc-connect/flink-cdc-source-connectors/flink-connector-postgres-cdc/src/main/java/com/ververica/cdc/connectors/postgres/source/fetch/PostgresSourceFetchTaskContext.java]: done it

**Review by leonardBang — APPROVED** [review]
Thanks @loserwang1024 for the update, LGTM

**Pull request merged**